### ISSUE TITLE
[Feature Request] IdMap/IdMap2 for IndexBinary

### ISSUE THREAD

**Issue by bduclaux** [facebookresearch/faiss#780]
Hello,

As far as I know, there is currently no way to use an IndexIDMap / IndexIDMap2 with binary indexes, as the IDMap classes derive from the Index class, and not the IndexBinary class.

Such code does not compile:
```
faiss::IndexBinaryHNSW * hnsw=new faiss::IndexBinaryHNSW(dim, 32);
faiss::IndexIDMap2 * hnsw_index=new faiss::IndexIDMap2(hnsw);
```
It would be nice to have IndexBinaryIDMap / IndexBinaryIDMap2 classes in a future release, with serialization support in index_io.cpp as for other indexes.

The current workaround is to store mappings between ids and record numbers in a separate binary file.

Best

**Comment by mdouze** [comment]
Yes we are progressively extending the meta-indexes to the binary indexes as well.

**Comment by mdouze** [comment]
Implemented in 1.5.3
